Process.c

```diff
@@ -179,35 +179,359 @@ void Process_fillStarttimeBuffer(Process* this) {
+    * component of the cmdline relative path, and retry the procedure. */
+   bool delimFound; /* if valid basename delimiter found */
+   do {
+      /* match basename */
+      matchLen = exeBaseLen + cmdlineBaseOffset;
+      if (cmdlineBaseOffset < exeBaseOffset &&
+          strncmp(cmdline + cmdlineBaseOffset, exe + exeBaseOffset, exeBaseLen) == 0)
+      {
+         delim = cmdline[matchLen];
+         if (delim == 0 || delim == '\n' || delim == ' ') {
+            int i, j;
+            /* reverse match the cmdline prefix and exe suffix */
+            for (i = cmdlineBaseOffset - 1, j = exeBaseOffset - 1;
+                 i >= 0 && j >= 0 && cmdline[i] == exe[j]; --i, --j)
+               ;
+
+            /* full match, with exe suffix being a valid relative path */
+            if (i < 0 && j >= 0 && exe[j] == '/')
+               return matchLen;
+         }
+      }
+
+      /* Try to find the previous potential cmdlineBaseOffset - it would be
+       * preceded by '/' or nothing, and delimited by ' ' or '\n' */
+      for (delimFound = false, cmdlineBaseOffset -= 2; cmdlineBaseOffset > 0; --cmdlineBaseOffset) {
+         if (delimFound) {
+            if (cmdline[cmdlineBaseOffset - 1] == '/') {
+               break;
+            }
+         } else if (cmdline[cmdlineBaseOffset] == ' ' || cmdline[cmdlineBaseOffset] == '\n') {
+            delimFound = true;
          }
-         len = this->basenameOffset - basename;
       }
+   } while (delimFound);
+
+   return 0;
+}
+
+/* stpcpy, but also converts newlines to spaces */
+static inline char *stpcpyWithNewlineConversion(char *dstStr, const char *srcStr) {
+   for (; *srcStr; ++srcStr) {
+      *dstStr++ = (*srcStr == '\n') ? ' ' : *srcStr;
+   }
+   *dstStr = 0;
+   return dstStr;
+}
+
+/*
+ * This function makes the merged Command string. It also stores the offsets of the
+ * basename, comm w.r.t the merged Command string - these offsets will be used by
+ * Process_writeCommand() for coloring. The merged Command string is also
+ * returned by Process_getCommandStr() for searching, sorting and filtering.
+ */
+void Process_makeCommandStr(Process *this) {
+   ProcessMergedCommand *mc = &this->mergedCommand;
+   const Settings *settings = this->settings;
+
+   bool showMergedCommand = settings->showMergedCommand;
+   bool showProgramPath = settings->showProgramPath;
+   bool searchCommInCmdline = settings->findCommInCmdline;
+   bool stripExeFromCmdline = settings->stripExeFromCmdline;
+
+   /* lp->mergedCommand.str needs updating only if its state or contents changed.
+    * Its content is based on the fields cmdline, comm, and exe. */
+   if (
+       mc->prevMergeSet == showMergedCommand &&
+       mc->prevPathSet == showProgramPath &&
+       mc->prevCommSet == searchCommInCmdline &&
+       mc->prevCmdlineSet == stripExeFromCmdline &&
+       !mc->cmdlineChanged &&
+       !mc->commChanged &&
+       !mc->exeChanged)
+   {
```

Comment:
Opening brace should be on previous line?

Reply:
Pondered about this, but I find it visually clearer to have the opening brace on a separate line. Otherwise this kinda looks strange.